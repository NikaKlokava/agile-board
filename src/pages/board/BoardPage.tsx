import { BoardContent, Footer, Header } from "./components";
import cl from "./board.module.css";
import { useEffect } from "react";
import { fetchBoards, fetchTasks } from "../../redux/thunk/fetchDataThunk";
import { useAppDispatch } from "../../redux/hooks/hook";

export const BoardPage = () => {
  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(fetchBoards());
    dispatch(fetchTasks());
  }, [dispatch]);

  const isLoading = false;

  return (
    <div className={cl.board}>
      <Header />
      <BoardContent isLoading={isLoading} />
      <Footer />
    </div>
  );
};
