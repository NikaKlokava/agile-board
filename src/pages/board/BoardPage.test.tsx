import renderer from "react-test-renderer";
import { BoardPage } from "./BoardPage";

describe("Test the BoardPage component", () => {
  test("The BoardPage renders correctly", () => {
    const boardPageSnap = renderer.create(<BoardPage />).toJSON();
    expect(boardPageSnap).toMatchSnapshot();
  });
});
