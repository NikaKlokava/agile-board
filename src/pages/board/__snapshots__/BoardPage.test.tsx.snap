// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the BoardPage component The BoardPage renders correctly 1`] = `
<div
  className="board"
>
  <div
    className="header_wrapper"
  >
    <div
      className="app_title"
    >
      <div
        className="app_logo"
      />
      <h1
        className="title"
      >
        AGILE-BOARD
      </h1>
      <p
        className="board_name"
      />
      <div
        className="navbar"
        onClick={[Function]}
      />
    </div>
    <div
      className="options"
    >
      <button
        className="button add-new-task"
        data-testid="add-new-task-btn"
        onClick={[Function]}
        type="button"
      >
        <div
          className="button_icon"
        />
        <div
          className="button_text"
        >
          Add New Task
        </div>
      </button>
      <div
        className="options_icon"
        data-testid="options-icon"
        onClick={[Function]}
      />
    </div>
  </div>
  <div>
    <div
      className="modal_wrapper"
      onClick={[Function]}
    />
    <div
      className="modal_container"
    >
       
      <form
        action="#"
        className="form_container"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <h2
          className="modal_title"
          data-testid="new-board-modal"
        >
          Add New Board
        </h2>
        <div
          className="container"
        >
          <p
            className="title"
          >
            Task name
          </p>
          <input
            autoComplete="off"
            data-testid="board-name-inpt"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="e.g Take coffee break"
            type="text"
            value=""
          />
        </div>
        <div
          className="container"
        >
          <p
            className="title"
          >
            Board Columns
          </p>
          <div
            className="input_container"
          >
            <input
              autoComplete="off"
              className="input_style"
              maxLength={14}
              name="columns[0]"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="e.g Take coffee break"
              type="text"
              value=""
            />
            <div
              className="delete_icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="input_container"
          >
            <input
              autoComplete="off"
              className="input_style"
              maxLength={14}
              name="columns[1]"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="e.g Take coffee break"
              type="text"
              value=""
            />
            <div
              className="delete_icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <button
          className="button"
          data-testid="add-new-column-btn"
          onClick={[Function]}
          type="button"
        >
          <div
            className="button_icon"
          />
          <div
            className="button_text"
          >
            Add New Column
          </div>
        </button>
        <button
          className="button center"
          data-testid="create-new-board-btn"
          type="submit"
        >
          <div
            className="button_text"
          >
            Create New Board
          </div>
        </button>
      </form>
    </div>
  </div>
  <div
    className="footer_wrapper"
  >
    <p
      className="footer_text"
    >
      Veranika Klokava
    </p>
    <p
      className="footer_text"
    >
      2023
    </p>
    <div
      className="footer_links"
    >
      <a
        href="https://instagram.com/nika_klokava?igshid=MmIzYWVlNDQ5Yg=="
      >
        <div
          className="instagram"
        />
      </a>
      <a
        href="https://linkedin.com/in/veranika-klokava-858b5b287"
      >
        <div
          className="linkedin"
        />
      </a>
      <a
        href="https://github.com/NikaKlokava"
      >
        <div
          className="github"
        />
      </a>
    </div>
  </div>
</div>
`;
